### ISSUE TITLE
`mut_mutex_lock` emitted on a `&mut &Mutex<T>`

### ISSUE THREAD

**Issue by dodomorandi** [rust-lang/rust-clippy#13201]
### Summary

The `mut_mutex_lock` is emitted when there is a unique reference to a shared reference of a mutex and the `lock` method is called. This is obviously a false positive.

### Lint Name

mut_mutex_lock

### Reproducer

I tried this code:

```rust
use std::sync::Mutex;

fn test(mutex: &mut &Mutex<i32>) {
    let _x = mutex.lock().unwrap();
}
```

I saw this happen:

```
warning: calling `&mut Mutex::lock` unnecessarily locks an exclusive (mutable) reference
 --> src/lib.rs:4:20
  |
4 |     let _x = mutex.lock().unwrap();
  |                    ^^^^ help: change this to: `get_mut`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#mut_mutex_lock
  = note: `#[warn(clippy::mut_mutex_lock)]` on by default
```

I expected to see this happen:

No lints are triggered.

### Version

```text
rustc 1.82.0-nightly (28a58f2fa 2024-07-31)
binary: rustc
commit-hash: 28a58f2fa7f0c46b8fab8237c02471a915924fe5
commit-date: 2024-07-31
host: x86_64-unknown-linux-gnu
release: 1.82.0-nightly
LLVM version: 19.1.0
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by ada4a** [comment]
This was fixed in https://github.com/rust-lang/rust-clippy/pull/13122 and no longer reproduces

@rustbot label S-fixed
